From 4114ee82f93c7471e804b4694e7bfa40fa350292 Mon Sep 17 00:00:00 2001
From: Julian Lehrhuber <jl@lerry.de>
Date: Mon, 10 Feb 2020 09:31:01 +0100
Subject: [PATCH 1/3] Modify default options and corresponding man page

Disable global announce
Disable relays
Disable nat
Disable usage reporting
Set urSeen to 3
---
 lib/config/optionsconfiguration.go | 12 ++++++------
 man/syncthing-config.5             | 20 ++++++++++----------
 2 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/lib/config/optionsconfiguration.go b/lib/config/optionsconfiguration.go
index d73426d2..23ce91a0 100644
--- a/lib/config/optionsconfiguration.go
+++ b/lib/config/optionsconfiguration.go
@@ -18,22 +18,22 @@ import (
 type OptionsConfiguration struct {
 	RawListenAddresses      []string `xml:"listenAddress" json:"listenAddresses" default:"default"`
 	RawGlobalAnnServers     []string `xml:"globalAnnounceServer" json:"globalAnnounceServers" default:"default"`
-	GlobalAnnEnabled        bool     `xml:"globalAnnounceEnabled" json:"globalAnnounceEnabled" default:"true"`
+	GlobalAnnEnabled        bool     `xml:"globalAnnounceEnabled" json:"globalAnnounceEnabled" default:"false"`
 	LocalAnnEnabled         bool     `xml:"localAnnounceEnabled" json:"localAnnounceEnabled" default:"true"`
 	LocalAnnPort            int      `xml:"localAnnouncePort" json:"localAnnouncePort" default:"21027"`
 	LocalAnnMCAddr          string   `xml:"localAnnounceMCAddr" json:"localAnnounceMCAddr" default:"[ff12::8384]:21027"`
 	MaxSendKbps             int      `xml:"maxSendKbps" json:"maxSendKbps"`
 	MaxRecvKbps             int      `xml:"maxRecvKbps" json:"maxRecvKbps"`
 	ReconnectIntervalS      int      `xml:"reconnectionIntervalS" json:"reconnectionIntervalS" default:"60"`
-	RelaysEnabled           bool     `xml:"relaysEnabled" json:"relaysEnabled" default:"true"`
+	RelaysEnabled           bool     `xml:"relaysEnabled" json:"relaysEnabled" default:"false"`
 	RelayReconnectIntervalM int      `xml:"relayReconnectIntervalM" json:"relayReconnectIntervalM" default:"10"`
 	StartBrowser            bool     `xml:"startBrowser" json:"startBrowser" default:"true"`
-	NATEnabled              bool     `xml:"natEnabled" json:"natEnabled" default:"true"`
+	NATEnabled              bool     `xml:"natEnabled" json:"natEnabled" default:"false"`
 	NATLeaseM               int      `xml:"natLeaseMinutes" json:"natLeaseMinutes" default:"60"`
 	NATRenewalM             int      `xml:"natRenewalMinutes" json:"natRenewalMinutes" default:"30"`
 	NATTimeoutS             int      `xml:"natTimeoutSeconds" json:"natTimeoutSeconds" default:"10"`
-	URAccepted              int      `xml:"urAccepted" json:"urAccepted"`                                    // Accepted usage reporting version; 0 for off (undecided), -1 for off (permanently)
-	URSeen                  int      `xml:"urSeen" json:"urSeen"`                                            // Report which the user has been prompted for.
+	URAccepted              int      `xml:"urAccepted" json:"urAccepted" default:"-1"`                       // Accepted usage reporting version; 0 for off (undecided), -1 for off (permanently)
+	URSeen                  int      `xml:"urSeen" json:"urSeen" default:"3"`                                // Report which the user has been prompted for.
 	URUniqueID              string   `xml:"urUniqueID" json:"urUniqueId"`                                    // Unique ID for reporting purposes, regenerated when UR is turned on.
 	URURL                   string   `xml:"urURL" json:"urURL" default:"https://data.syncthing.net/newdata"` // usage reporting URL
 	URPostInsecurely        bool     `xml:"urPostInsecurely" json:"urPostInsecurely" default:"false"`        // For testing
@@ -56,7 +56,7 @@ type OptionsConfiguration struct {
 	SetLowPriority          bool     `xml:"setLowPriority" json:"setLowPriority" default:"true"`
 	RawMaxFolderConcurrency int      `xml:"maxFolderConcurrency" json:"maxFolderConcurrency"`
 	CRURL                   string   `xml:"crashReportingURL" json:"crURL" default:"https://crash.syncthing.net/newcrash"` // crash reporting URL
-	CREnabled               bool     `xml:"crashReportingEnabled" json:"crashReportingEnabled" default:"true"`             // Read in the monitor, but it's read before every attempt to report stuff, so does not require a restart.
+	CREnabled               bool     `xml:"crashReportingEnabled" json:"crashReportingEnabled" default:"false"`            // Read in the monitor, but it's read before every attempt to report stuff, so does not require a restart.
 	StunKeepaliveStartS     int      `xml:"stunKeepaliveStartS" json:"stunKeepaliveStartS" default:"180"`                  // 0 for off
 	StunKeepaliveMinS       int      `xml:"stunKeepaliveMinS" json:"stunKeepaliveMinS" default:"20"`                       // 0 for off
 	RawStunServers          []string `xml:"stunServer" json:"stunServers" default:"default"`
diff --git a/man/syncthing-config.5 b/man/syncthing-config.5
index 12a61130..ddfa4de6 100644
--- a/man/syncthing-config.5
+++ b/man/syncthing-config.5
@@ -124,22 +124,22 @@ The following shows an example of the default configuration file (IDs will diffe
         <listenAddress>tcp://0.0.0.0:8384</listenAddress>
         <listenAddress>dynamic+https://relays.syncthing.net/endpoint</listenAddress>
         <globalAnnounceServer>default</globalAnnounceServer>
-        <globalAnnounceEnabled>true</globalAnnounceEnabled>
+        <globalAnnounceEnabled>false</globalAnnounceEnabled>
         <localAnnounceEnabled>true</localAnnounceEnabled>
         <localAnnouncePort>21027</localAnnouncePort>
         <localAnnounceMCAddr>[ff12::8384]:21027</localAnnounceMCAddr>
         <maxSendKbps>0</maxSendKbps>
         <maxRecvKbps>0</maxRecvKbps>
         <reconnectionIntervalS>60</reconnectionIntervalS>
-        <relaysEnabled>true</relaysEnabled>
+        <relaysEnabled>false</relaysEnabled>
         <relayReconnectIntervalM>10</relayReconnectIntervalM>
         <startBrowser>true</startBrowser>
-        <natEnabled>true</natEnabled>
+        <natEnabled>false</natEnabled>
         <natLeaseMinutes>60</natLeaseMinutes>
         <natRenewalMinutes>30</natRenewalMinutes>
         <natTimeoutSeconds>10</natTimeoutSeconds>
-        <urAccepted>0</urAccepted>
-        <urSeen>0</urSeen>
+        <urAccepted>-1</urAccepted>
+        <urSeen>3</urSeen>
         <urUniqueID></urUniqueID>
         <urURL>https://data.syncthing.net/newdata</urURL>
         <urPostInsecurely>false</urPostInsecurely>
@@ -772,22 +772,22 @@ Skip verification (true or false).
     <listenAddress>tcp://0.0.0.0:8384</listenAddress>
     <listenAddress>dynamic+https://relays.syncthing.net/endpoint</listenAddress>
     <globalAnnounceServer>default</globalAnnounceServer>
-    <globalAnnounceEnabled>true</globalAnnounceEnabled>
+    <globalAnnounceEnabled>false</globalAnnounceEnabled>
     <localAnnounceEnabled>true</localAnnounceEnabled>
     <localAnnouncePort>21027</localAnnouncePort>
     <localAnnounceMCAddr>[ff12::8384]:21027</localAnnounceMCAddr>
     <maxSendKbps>0</maxSendKbps>
     <maxRecvKbps>0</maxRecvKbps>
     <reconnectionIntervalS>60</reconnectionIntervalS>
-    <relaysEnabled>true</relaysEnabled>
+    <relaysEnabled>false</relaysEnabled>
     <relayReconnectIntervalM>10</relayReconnectIntervalM>
     <startBrowser>true</startBrowser>
-    <natEnabled>true</natEnabled>
+    <natEnabled>false</natEnabled>
     <natLeaseMinutes>60</natLeaseMinutes>
     <natRenewalMinutes>30</natRenewalMinutes>
     <natTimeoutSeconds>10</natTimeoutSeconds>
-    <urAccepted>0</urAccepted>
-    <urSeen>0</urSeen>
+    <urAccepted>-1</urAccepted>
+    <urSeen>3</urSeen>
     <urUniqueID></urUniqueID>
     <urURL>https://data.syncthing.net/newdata</urURL>
     <urPostInsecurely>false</urPostInsecurely>
-- 
2.24.0.windows.2

